pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -527,14 +530,15 @@ def valid_items(self) -> Generator:
     def _forked_name(self, result_metric: ResultMetric, on_step: bool) -> Tuple[str, str]:
         name = result_metric.meta.name
         forked_name = result_metric.meta.forked_name(on_step)
+        add_datalaoder_idx = result_metric.meta.add_dataloader_idx
         dl_idx = result_metric.meta.dataloader_idx
-        if dl_idx is not None:
+        if add_datalaoder_idx and dl_idx is not None:
```

Comment:
This is an implementation detail from the loops leaking into the resultcollection, who is the `dl_idx is not None` check enough?

Reply:
we still pass in correct dataloader_idx regardless of `add_dataloader_idx. See the example in the linked issue.